src/App.js

```diff
@@ -14,11 +15,88 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    goods: goodsFromServer.map(good => ({
+      name: good, selected: false,
+    })),
+  };
+
+  componentDidMount() {
+    this.changeStatus(8);
+  }
+
+  changeStatus = (index) => {
+    this.setState((state) => {
+      const copy = [...state.goods];
+
+      copy[index].selected = !copy[index].selected;
+
+      return { goods: copy };
+    });
+  };
```

Comment:
Я не до конца уверен, но мне кажется, что тут не обязательно делать колбек и принимать прошлый state. 
А просто вот так работать не будет?
```suggestion
  changeStatus = (index) => {
    this.setState({
      goods[index].selected = !goods[index].selected;
    });
  };
```

Reply:
Це не буде працювати.

Reply:
When you change state relying on the previous state you shoud use callback